cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h

```diff
@@ -378,7 +372,22 @@ class SPCORE_API Unreal
         }
     #endif
 
-    static std::string getStableComponentName(const USceneComponent* scene_component, bool include_stable_actor_name = false)
+    static std::string getStableComponentName(const CActorComponent auto* actor_component, bool include_stable_actor_name = false)
+    {
+        SP_ASSERT(actor_component);
+
+        std::string component_name = toStdString(actor_component->GetName());
+
+        if (include_stable_actor_name) {
+            AActor* actor = actor_component->GetOwner();
+            SP_ASSERT(actor);
+            component_name = getStableActorName(actor) + ":" + component_name;
+        }
+
+        return component_name;
+    }
+
+    static std::string getStableComponentName(const CSceneComponent auto* scene_component, bool include_stable_actor_name = false)
```

Comment:
I think it's a little confusing to have this function called `getStableComponentName` - when I first read this, it felt like I was getting the `StableName` of a component. Of course, we don't add `StableNameComponent` to a component, so it'd be wrong to think from that angle, but it's a bit confusing IMO. 

Why not just call this function `getComponentName`, and the boolean as `include_actor_name`? I think it'd be a little less confusing (i.e., less confusion between the meaning of `Stable` here vs `StableNameComponent`) with no `Stable` word.

Reply:
To me, the _stable name_ of an entity in Unreal means the unique string that we use throughout all SPEAR code (i.e., editor-only Python code, C++ code, MuJoCo code, etc) to refer to that entity. I like using the term "stable name" here to distinguish this concept from all of the other similar concepts in the Unreal Engine (`getName()`, `getActorLabel()`, etc).

When you call this function, you are getting the _stable name_ of the component, in the sense that you're getting the unique string that we use throughout all SPEAR code to refer to that component.

If we called this function `getComponentName`, I think it would be more confusing, because I don't know what underlying naming concept is being used, among all the following possibilities, all of which would be reasonable guesses:

```
actor->getName() + component->getName()
actor->getActorLabel() + component->getName()
root_component->getName() + child_component->getName() + grandchild_component->getName()
getStableActorName(actor) + component->getName()
...
```

I like calling this function `getStableComponentName` because a "stable name" has a specific meaning in SPEAR, so there is no guesswork to do to know what this function is going to return.

The fact that you need to add exactly one `StableNameComponent` to each actor in order to implement the `getStableActorName(actor)` function is an implementation detail. The fact that components don't need anything like that in order to implement the `getStableComponentName(component)` function is also an implementation detail. I don't think either implementation detail it should drive the naming conventions for these functions.